src/core/utils/channel_manager.hpp

```diff
@@ -0,0 +1,193 @@
+ */
+
+#ifndef CHANNEL_MANAGER_HPP_
+#define CHANNEL_MANAGER_HPP_
+
+#include "openthread-core-config.h"
+
+#include <openthread/platform/radio.h>
+#include <openthread/types.h>
+
+#include "common/locator.hpp"
+#include "common/notifier.hpp"
+#include "common/timer.hpp"
+
+namespace ot {
+namespace Utils {
+
+/**
+ * @addtogroup utils-channel-manager
+ *
+ * @brief
+ *   This module includes definitions for Channel Manager.
+ *
+ * @{
+ */
+
+#if OPENTHREAD_ENABLE_CHANNEL_MANAGER
+
+#if OPENTHREAD_FTD
+
+/**
+ * This class implements the Channel Manager.
+ *
+ */
+class ChannelManager : public InstanceLocator
+{
+public:
+    enum
+    {
+        /**
+         * Minimum delay in seconds used for network channel change.
+         *
+         */
+        kMinimumDelay = OPENTHREAD_CONFIG_CHANNEL_MANAGER_MINIMUM_DELAY,
+    };
+
+    /**
+     * This constructor initializes a `ChanelManager` object.
+     *
+     * @param[in]   aInstance  A reference to the OpenThread instance.
+     *
+     */
+    explicit ChannelManager(Instance &aInstance);
+
+    /**
+     * This method requests a Thread network channel change.
+     *
+     * The Thread network switches to the given channel after a specified delay (@sa GetDelay()). The channel change is
+     * performed by updating the Pending Operational Dataset.
+     *
+     * A subsequent call to this method will cancel an ongoing previously requested channel change.
+     *
+     * @param[in] aChannel             The new channel for the Thread network.
+     *
+     * @retval OT_ERROR_NONE           Channel change request successfully processed.
+     * @retval OT_ERROR_INVALID_ARGS   The new channel is not a supported channel.
+     *
+     */
+    otError RequestChannelChange(uint8_t aChannel);
+
+    /**
+     * This method gets the  delay (in seconds) used for a channel change.
+     *
+     * @returns The delay (in seconds)
+     *
+     */
+    uint16_t GetDelay(void) const { return mDelay; }
+
+    /**
+     * This method sets the delay (in seconds) used for a channel change.
+     *
+     * The delay should preferably be longer than maximum data poll interval used by all sleepy-end-devices within the
+     * Thread network.
+     *
+     * @param[in]  aDelay             Delay in seconds.
+     *
+     * @retval OT_ERROR_NONE          Delay was updated successfully.
+     * @retval OT_ERROR_INVALID_ARGS  The given delay @p aDelay is shorter than `kMinimumDelay`.
+     *
+     */
+    otError SetDelay(uint16_t aDelay);
+
+    /**
+     * This method gets the supported channel mask.
+     *
+     * @returns  The supported channels mask.
+     *
+     */
+    uint32_t GetSupportedChannels(void) const { return mSupportedChannels; }
+
+    /**
+     * This method sets the supported channel mask.
+     *
+     * @param[in]  aChannelMask  A channel mask.
+     *
+     */
+    void SetSupportedChannels(uint32_t aChannelMask) {
+        mSupportedChannels = (aChannelMask | OT_RADIO_SUPPORTED_CHANNELS);
```

Comment:
Is this supposed to be bit-wise AND?

Reply:
Yes! Indeed! It's not correct... Thanks man!